instrumentation/ruby_kafka/test/opentelemetry/instrumentation/ruby-kafka/patches/consumer_test.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+require_relative '../../../../../lib/opentelemetry/instrumentation/ruby_kafka'
+require_relative '../../../../../lib/opentelemetry/instrumentation/ruby_kafka/patches/consumer'
+
+describe OpenTelemetry::Instrumentation::RubyKafka::Patches::Consumer do
+  let(:instrumentation) { OpenTelemetry::Instrumentation::RubyKafka::Instrumentation.instance }
+  let(:exporter) { EXPORTER }
+  let(:spans) { exporter.finished_spans }
+
+  let(:host) { ENV.fetch('TEST_KAFKA_HOST') { '127.0.0.1' } }
+  let(:port) { (ENV.fetch('TEST_KAFKA_PORT') { 29_092 }) }
+
+  let(:kafka) { Kafka.new(["#{host}:#{port}"], client_id: 'opentelemetry-kafka-test') }
+  let(:topic) { "topic-#{SecureRandom.uuid}" }
+  let(:producer) { kafka.producer }
+  let(:consumer) { kafka.consumer(group_id: SecureRandom.uuid, fetcher_max_queue_size: 1) }
+
+  before do
+    kafka.create_topic(topic)
+    consumer.subscribe(topic)
+
+    # Clear spans
+    exporter.reset
+
+    instrumentation.install
+  end
+
+  after do
+    # Force re-install of instrumentation
+    instrumentation.instance_variable_set(:@installed, false)
+
+    # Clean up
+    producer.shutdown
+    consumer.stop
+    kafka.close
+  end
+
+  describe '#each_message' do
+    it 'traces each_meassage call' do
+      kafka.deliver_message('hello', topic: topic)
+      kafka.deliver_message('hello2', topic: topic)
+
+      begin
+        @counter = 0
```

Comment:
What is this counter for? It seems like a hold over from an earlier iteration of the code.